src/systems/drawing-menu-system.js

```diff
@@ -0,0 +1,200 @@
+import { waitForDOMContentLoaded } from "../utils/async-utils";
+
+/**
+ * Drawing Menu System
+ * A tool that allows drawing on networked-drawing components.
+ * @namespace drawing
+ * @component pen
+ */
+
+function almostEquals(epsilon, u, v) {
+  return Math.abs(u.x - v.x) < epsilon && Math.abs(u.y - v.y) < epsilon && Math.abs(u.z - v.z) < epsilon;
+}
+
+export class DrawingMenuSystem {
+  constructor(sceneEl) {
+    this.sceneEl = sceneEl;
+
+    this.sceneEl.addEventListener("stateremoved", this.hideMenus.bind(this));
+
+    this.lastIntersection = new THREE.Vector3();
+
+    this.drawingMenus = [];
+    this.setDirty = this.setDirty.bind(this);
+    this.dirty = true;
+
+    // TODO: Use the MutationRecords passed into the callback function to determine added/removed nodes!
+    this.observer = new MutationObserver(this.setDirty);
+
+    waitForDOMContentLoaded().then(() => {
+      this.cursorControllers = document.querySelectorAll("[cursor-controller]");
+      this.camera = this.sceneEl.querySelector("#avatar-pov-node");
+      this.observer.observe(this.sceneEl, { childList: true, attributes: true, subtree: true });
+      this.sceneEl.addEventListener("object3dset", this.setDirty.bind(this));
+      this.sceneEl.addEventListener("object3dremove", this.setDirty.bind(this));
+    });
+  }
+
+  tick() {
+    if (!this.cursorControllers || this.cursorControllers.length === 0 || !this.camera) return;
+
+    if (this.sceneEl.is("frozen")) {
+      if (this.dirty) {
+        this.populateEntities(this.targets);
+        this.dirty = false;
+      }
+
+      const hovered = this.getHovered();
+      if (hovered) {
+        for (let i = 0; i < this.cursorControllers.length; i++) {
+          if (this.cursorControllers[i].components["cursor-controller"].intersection) {
+            const intersectionPoint = this.cursorControllers[i].components["cursor-controller"].intersection.point;
+            this.showMenu(hovered, intersectionPoint);
```

Comment:
Can this end up showing the menu twice, or on the wrong intersection point?

Reply:
Oh, I think you need to pull master into this branch to test it with the left hand changes

Reply:
since you can't have two cursors active at the same time (at least while in pause mode) this shouldn't be an issue, but even if you could, it would just move the menu to whichever cursor is second in the list (left-cursor I think), and only if that cursor is at least a certain distance away from the first cursor.